### PR TITLE
Generator: Parameterized delete

### PR DESCRIPTION
**Pull request by masnax**
The goal here is adding explicit arguments to delete, and introducing a few helpers that will come in handy later on as we refactor/change more of the generator.

Main changes: 
* Generated Delete methods now take arguments instead of filters.
* Generating Delete methods now requires arguments to be passed to the generator via -by- statements, as in: `DeleteMany-by-Name-and-Type`.
* Added `ActiveFilters` and `activeFilters` helpers for parsing the `-by-` field into slices of `Field` and `string`, respectively. 
* Added proper handling of fields named `Type` to `FieldArgs` and `FieldParams` by making the functions into methods on `Mapper`. As well, added support for adding additional arguments directly into the methods, so all arguments are handled in one place.

### CODE REVIEW

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
Re-running tests as I was reconfiguring subnets in the middle of the previous run.
Looking at the resulting code and changes outside of .mapper.go, this looks good to me.

But I'll let @tomponline do a final review and merge if he's happy with it.

File: lxd/db/operations.mapper.go
Comment (tomponline) [inline lxd/db/operations.mapper.go:150]: @masnax something odd with the argument name here.

**Review by tomponline — APPROVED** [review]
I'll merge this, but if you can send up another small PR to fix that `uUID` argument name that would be great. Thanks

**Pull request merged**
